samples/interactive-tutorials/user_environment_setup.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Copyright 2022 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# set the Google Cloud project Id
+project_id=$1
+echo Project ID: $project_id
+gcloud config set project project_id
+
+timestamp=$(date +%s)
+
+service_account_id="service-acc-"$timestamp
+echo Service Account: $service_account_id
+
+# create service account (your project_id+timestamp)
+gcloud iam service-accounts create $service_account_id
+
+# assign needed roles to your new service account
+for role in {retail.admin,editor,bigquery.admin}
+  do
+    gcloud projects add-iam-policy-binding $project_id --member="serviceAccount:"$service_account_id"@"$project_id".iam.gserviceaccount.com" --role="roles/${role}"
```

Comment:
You should be okay to surround the entire string including the variables with quotes. You won't need to split the quotes between non variable and variable portions.
`"serviceAccount:$service_account_id@$project_id.iam.gserviceaccount.com"`

Reply:
fixed